core/DataTable/Filter/ColumnCallbackAddMetadata.php

```diff
@@ -45,10 +45,6 @@ public function __construct($table, $columnsToRead, $metadataToAdd, $functionToA
     {
         parent::__construct($table);
 
-        if (!is_array($columnsToRead)) {
-            $columnsToRead = array($columnsToRead);
-        }
-
         $this->columnsToRead      = $columnsToRead;
```

Comment:
I don't like the idea of explicitly including redundant code in core. If this is not important to the optimizations, I would suggest moving it to a new issue where someone can hopefully find a reason for the speed increase & find a clean fix. I'll leave it to @mattab to decide whether it's worth it to include this debt now.